src/main/java/net/greghaines/jesque/worker/WorkerImpl.java

```diff
@@ -529,21 +533,40 @@ protected void process(final Job job, final String curQueue) {
             final Object instance = this.jobFactory.materializeJob(job);
             final Object result = execute(job, curQueue, instance);
             success(job, instance, result, curQueue);
+        } catch (DontPerformException e) {
+            try {
+                Thread.sleep(dontPerformSleepTime);
```

Comment:
Yes, you are right. I will redo the patch. The test failure seems to be a timing issue. But when redoing the patch, I have to redo the test anyway.